src/backend/CassandraBackendNew.h

```diff
@@ -488,6 +488,81 @@ class BasicCassandraBackend : public BackendInterface
         return {txns, {}};
     }
 
+    NFTsAndCursor
+    fetchNFTsByIssuer(
+        ripple::AccountID const& issuer,
+        std::optional<std::uint32_t> const& taxon,
+        std::uint32_t const ledgerSequence,
+        std::uint32_t const limit,
+        std::optional<ripple::uint256> const& cursorIn,
+        boost::asio::yield_context& yield) const override
+    {
+        NFTsAndCursor ret;
+
+        Statement idQueryStatement = [&taxon, &issuer, &cursorIn, &limit, this]() {
+            if (taxon.has_value())
+            {
+                auto r = schema_->selectNFTIDsByIssuerTaxon.bind(issuer);
+                r.bindAt(1, *taxon);
+                r.bindAt(2, cursorIn.value_or(ripple::uint256(0)));
+                r.bindAt(3, limit);
+                return r;
+            }
+
+            auto r = schema_->selectNFTIDsByIssuer.bind(issuer);
+            r.bindAt(
+                1,
+                std::make_tuple(
+                    cursorIn.has_value() ? ripple::nft::toUInt32(ripple::nft::getTaxon(*cursorIn)) : 0,
+                    cursorIn.value_or(ripple::uint256(0))));
+            r.bindAt(2, limit);
+            return r;
+        }();
+
+        auto const& idQueryResults = executor_.read(yield, idQueryStatement).value();
+        std::vector<ripple::uint256> nftIDs;
+        for (auto const [nftID] : extract<ripple::uint256>(idQueryResults))
+            nftIDs.push_back(nftID);
+
+        if (nftIDs.size() == 0)
+            return ret;
+
+        if (nftIDs.size() == limit)
+            ret.cursor = nftIDs.back();
+
+        /// TODO these two queries should happen in parallel
+        auto nftQueryStatement = schema_->selectNFTBulk.bind(nftIDs);
+        nftQueryStatement.bindAt(1, ledgerSequence);
+
+        auto const& nftQueryResults = executor_.read(yield, nftQueryStatement).value();
+
+        auto nftURIQueryStatement = schema_->selectNFTURIBulk.bind(nftIDs);
+        nftURIQueryStatement.bindAt(1, ledgerSequence);
+
+        auto const& nftURIQueryResults = executor_.read(yield, nftURIQueryStatement).value();
+        ///
+        //
+
+        std::unordered_map<std::string, Blob> nftURIMap;
+        for (auto const [nftID, uri] : extract<ripple::uint256, Blob>(nftURIQueryResults))
+            nftURIMap.insert({ripple::strHex(nftID), uri});
```

Comment:
note - this is not ideal to have the keys be `std::string` instead of `ripple::uint256` but it seems to me that the latter has no implemented hash function so it cannot be used as the key to a map.

Reply:
You could use a `std::map` instead. I assume we don't expect millions of URIs.